downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1562 @@
+- This is an inter-dependent module for the `websub` module.
+
+**Hub Implementation**
+
+- The default `websub:Hub` implementation has been removed and language-specific API abstraction is defined in the `websubhub:Service`.
+- Updated the WebSub Hub sample implementation to comply with the new API specification as follows.
+
+    ```ballerina
+    listener websubhub:Listener hubListener = new (9001);
+
+    service / websubhubon hubListener {
+
+    remote function onRegisterTopic(TopicRegistration message) returns TopicRegistrationSuccess|TopicRegistrationError {
+        // implement action to execute on topic-registration
+        }
+
+    remote function onDeregisterTopic(TopicDeregistration message) returns TopicDeregistrationSuccess|TopicDeregistrationError {
+        // implement action to execute on topic-deregistration
+        }
+
+    remote function onUpdateMessage(UpdateMessage msg) returns Acknowledgement|UpdateMessageError {
+        // implement action to execute on content-update for topic
+        }
+
+    remote function onSubscription(Subscription msg) returns SubscriptionAccepted|SubscriptionPermanentRedirect|
+        SubscriptionTemporaryRedirect|BadSubscriptionError|InternalSubscriptionError {
+        // implement action to execute on new subscription
+        }
+
+    remote function onSubscriptionValidation(Subscription msg) returns SubscriptionDeniedError? {
+        // implement action to execute on subscription validation
+        }
+
+    remote function onSubscriptionIntentVerified(VerifiedSubscription msg) {
+        // implement action to execute on subscription intent verification
+        }
+
+    remote function onUnsubscription(Unsubscription msg) returns UnsubscriptionAccepted|BadUnsubscriptionError|
+        InternalUnsubscriptionError {
+        // implement action to execute on unsubscription
+        }
+
+    remote function onUnsubscriptionValidation(Unsubscription msg) returns UnsubscriptionDeniedError? {
+        // implement action to execute on unsubscription validation
+        }
+
+    remote function onUnsubscriptionIntentVerified(VerifiedUnsubscription msg) {
+        // implement action to execute on unsubscription intent verification
+        }
+    }    
+    ```
+
+**Hub Client Implementation**
+
+- The `websubhub:Client` is introduced to distribute the updated content to subscribers.
+- The example below is a sample use-case of the WebSub Hub Client.
+
+```ballerina
+    service / websubhubon hubListener {
+    remote function onSubscriptionIntentVerified(websubhub:VerifiedSubscription msg) {
+
+            // you can pass client config if you want 
+            // say maybe retry config
+            websub:HubClient hubclient = new (msg);
+            check start notifySubscriber(hubclient);
+        }
+
+        function notifySubscriber(websubhub:HubClient hubclient) returns error? {
+            while (true) {
+                // fetch the message from MB
```

Comment:
IMO MB here is out of context, and the user may not even use an MB.

Can we just say
```suggestion
                // Fetch the messages to be delivered. 
```

Reply:
@ayeshLK